### DISCUSSION TITLE
Add ability to reorder Line Items when creating/editing CP Orders

### DISCUSSION THREAD

**Discussion by BrandKits** [craftcms/commerce#3597] [Ideas]
### What happened?

There's currently no way to reorder line items in the CP. 

If you create an order... and forget to add a line item, it automatically appears at the top when added, with no way to reorder items, jumbling the order, and making little sense to the customer.

Can Order > Line Items have the same controls as Categories/Structures, reflecting the order on the back and front ends?

### Craft CMS version

3.8.1

### Craft Commerce version

3.4.20

### Installed plugins and versions

-

**Comment by jelmer-scoremedia** [comment]
I like this

**Comment by lukeholder** [comment]
Great idea, added this as a discussion. We welcome more feedback on this.

**Reply by BrandKits** [reply]
Thanks Luke, I originally added this as a discussion https://github.com/craftcms/commerce/discussions/2792 but it got no response. I then lost the client as they required this ability, so bumped it up to an issue. It would be great to see this implemented.

**Comment by sjcallender** [comment]
We have a use case for this. We have a custom discount/adjuster that allows the first 4 of the product type to be free. As customers add more than 4, the specific purchasables that are marked as free switch to the be the newly added ones since the whole cart is recalculated with each new quantity or item change. It's a mildly jarring UX.

**Reply by piotrpog** [reply]
I have a client that requested it, but not as a manual reorder - in this case it would be automatic one, presumably using some kind of event. Sorted line items would then retain new order every time we call `order->lineItems` - so not only in the order page in the control panel, but also on the fronted templates or order pdfs.

This is how imagine it would work, for sorting line items by their title:

```
        Event::on(
            Order::class,
            Element::EVENT_BEFORE_SAVE,
            function(ModelEvent $e) {
                $order = $e->sender;
                $lineItems = $e->sender->lineItems;
                usort($lineItems, function($a, $b) {
                    return strcmp($a->getPurchasable()->product->title, $b->getPurchasable()->product->title);
                });
                $order->setLineItems($lineItems);
            }
        );
```

My client was actually requesting sorting items by their category title, but code above could be easily adjusted to also do that.

Could we expect adding this feature in the upcoming versions of the commerce?

**Reply by lukeholder** [reply]
Its on our list but we cant give a timeline for it as the moment. We will monitor this discussion and reassess as needed. Thanks.
